auth/client/src/main/java/org/wildfly/security/auth/client/AuthenticationConfiguration.java

```diff
@@ -748,6 +748,23 @@ public AuthenticationConfiguration usePassword(Password password) {
         return password == null ? useCredentials(filtered) : useCredentials(filtered).useCredential(new PasswordCredential(password));
     }
 
+    /**
+     * Create a new configuration which is the same as this configuration, but which uses the given encrypted
+     * password to authenticate. The current encryption client configuration is loaded and is used to decrypt
+     * the encrypted password. If one does not exist, appropriate exception is thrown.
+     *
+     * @param encryptedPassword the password to use
```

Comment:
I added this function to decrypt a password before adding it so if users want to programmatically configure an auth client with encrypted expressions, they can do that.

To make it more general, we can replace it with a function called `decryptAndUseValue()` that takes in the expression to decrypt and the function to call afterwards. However, I am not sure if that is something we can do with templates (not sure if that's what they are called in java. But here is what I am talking about https://cplusplus.com/doc/oldtutorial/templates/). 

If this is not supported, then we can send enumerated values like [this](https://github.com/wildfly-security/wildfly-elytron/blob/2.x/auth/client/src/main/java/org/wildfly/security/auth/client/AuthenticationConfiguration.java#L273-L300) function is doing. So, based on the enum value, we can call the appropriate function with the decrypted value. 

@Skyllarr @fjuma please let me know what you think.